### PR TITLE
Hca 771 search endpoint bugs

### PR DESCRIPTION
**Pull request by rolando-ebi**

### CODE REVIEW

File: src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeRepository.java [hunk 1]
```diff
@@ -18,13 +20,16 @@
 @CrossOrigin
 public interface SubmissionEnvelopeRepository extends MongoRepository<SubmissionEnvelope, String> {
 
-    @RestResource(path="findByUuid", rel="findByUuid")
+    @RestResource(exported = false)
```
Comment (tburdett) [inline src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeRepository.java]: Why do you _not_ want to export this? Isn't this potentially useful?

Comment (rolando-ebi) [reply src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeRepository.java]: It's only used internally in the core and requires the nested Uuid{UUID} object as a the search param

**Comment by tburdett** [comment]
I general, I'm wondering whether we've done a review over the usage of the `@Param` annotation or whether we are compiling our code with the -param flag.  If we've done neither, this bug will pop up everywhere.

File: src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeRepository.java (hunk 1 above)
Comment (tburdett) [reply src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeRepository.java]: Ok, that's good.  Would it therefore be worth exporting the `findbyUuidUUID` method with a cleaner name?

Comment (rolando-ebi) [reply src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeRepository.java]: It's exported below as "findByUuid", but does a "findByUuidUuid" under the hood

Comment (tburdett) [reply src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeRepository.java]: Aha! Looking at the wrong view! Thanks 👍

**Comment by rolando-ebi** [comment]
compiling with -parameters in the build.gradle now: https://github.com/HumanCellAtlas/ingest-core/commit/bc76aea7f9f7c66f17124faeb1dd01118d42a197

**Pull request merged**